src/event.rs

```diff
@@ -546,8 +568,25 @@ impl<N: Notify> Processor<N> {
             window.hide();
         }
 
-        if self.window_changes.hide {
-            window.hide();
+        #[cfg(target_os = "macos")]
+        {
+            if self.window_changes.toggle_fullscreen && !self.is_simple_fullscreen {
+                window.set_fullscreen(!self.is_fullscreen);
+                self.is_fullscreen = !self.is_fullscreen;
+            }
+
+            if self.window_changes.toggle_simple_fullscreen && !self.is_fullscreen {
+                window.set_simple_fullscreen(!self.is_simple_fullscreen);
+                self.is_simple_fullscreen = !self.is_simple_fullscreen;
+            }
+        }
+
+        #[cfg(not(target_os = "macos"))]
+        {
+            if self.window_changes.toggle_fullscreen {
+                window.set_fullscreen(!self.is_fullscreen);
+                self.is_fullscreen = !self.is_fullscreen;
+            }
         }
```

Comment:
I think I've mentioned this in a previous comment, but we should aim to merge the macos and non-macos targets together here, after all the body of this conditional is completely identical.

The easiest solution would probably be to just make `is_simple_fullscreen` a field that is available on every platform and just make it always false on linux/bsd/windows.

Reply:
:+1: